plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -292,10 +297,21 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 			continue
 		}
 		secretName := source.Secret.SecretName
-		if !referencedSecrets.Has(secretName) {
+		if !mountableSecrets.Has(secretName) {
```

Comment:
In https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/security.md#description-of-roles I would say @lavalamp's role is "developer", and it's up to the project admin to decide whether the developer can use any secret, or just those isolated by service accounts.  In the future, that may mean that LimitSecrets is managed per namespace.  Right now, it seems like that the only issue is the use of the secrets (not the design of service accounts or secrets), so the choice of LimitSecrets encourages a style.  If your project admin allows you to change service accounts, it's not an issue.  If a project admin doesn't want you to change service accounts, then it's because they don't want you managing secrets directly and they likely don't let you change secrets (just read them).